catalyst-gateway/bin/src/db/event/config/key.rs

```diff
@@ -0,0 +1,90 @@
+//! Configuration Key
+
+use std::net::IpAddr;
+
+use serde_json::Value;
+
+/// Configuration key
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum ConfigKey {
+    /// Frontend general configuration
+    Frontend,
+    /// Frontend configuration for a specific IP address
+    FrontendForIp(IpAddr),
+}
+
+/// Frontend JSON schema
+const FRONTEND_JSON_SCHEMA: &str = include_str!("jsonschema/frontend.json");
+/// Default frontend configuration
+const FRONTEND_DEFAULT: &str = include_str!("default/frontend.json");
+
+impl ConfigKey {
+    /// Create a `ConfigKey` from ids.
+    pub(crate) fn from_id(id1: &str, id2: &str, id3: &str) -> Option<Self> {
+        match (id1, id2, id3) {
+            ("frontend", "", "") => Some(ConfigKey::Frontend),
+            ("frontend", "ip", ip) => ip.parse().ok().map(ConfigKey::FrontendForIp),
+            _ => None,
+        }
+    }
+
+    /// Convert a `ConfigKey` to its ids.
+    pub(super) fn to_id(&self) -> (String, String, String) {
+        match self {
+            ConfigKey::Frontend => ("frontend".to_string(), String::new(), String::new()),
+            ConfigKey::FrontendForIp(ip) => {
+                ("frontend".to_string(), "ip".to_string(), ip.to_string())
+            },
+        }
+    }
+
+    /// Validate the provided value against the JSON schema.
+    pub(super) fn validate(&self, value: &Value) -> anyhow::Result<()> {
+        let schema: Value = serde_json::from_str(FRONTEND_JSON_SCHEMA)
+            .map_err(|e| anyhow::anyhow!("Failed to parse JSON schema: {:?}", e))?;
+
+        let validator = jsonschema::validator_for(&schema)
+            .map_err(|e| anyhow::anyhow!("Failed to create JSON validator: {:?}", e))?;
+
```

Comment:
All of the previous should be done once.
Use a lazy lock, so we are not continuously re-parsing the json schema from the sting.
Pattern for using static deserialization/parsing of constant strings: https://stackoverflow.com/questions/58359340/deserialize-file-using-serde-json-at-compile-time

But use lazy lock from standard library:
https://doc.rust-lang.org/beta/std/sync/struct.LazyLock.html

Then the match should simply assign the static evaluated schema to the  `validator`.  variable.

If there is an error evauating the schema, log an error, and default to a fully permissive default schema.